src/plots/plots.js

```diff
@@ -1158,9 +1159,21 @@ plots.supplyTraceDefaults = function(traceIn, traceOut, colorIndex, layout, trac
         }
     }
 
+    function coerceUnlessPruned(attr, dflt, cb) {
```

Comment:
We probably don't need this now that `customdata` and `ids` are deemed ok. What if we just add a `noHover` category to handle the `fx.supplyDefaults` call below. Oh well, @antoinerg already got a :dancer: , so I guess I'm a little late to the party.

Reply:
But really, at the very least we should add a test check that `parcoords` traces don't have `hoverlabel` in their fullData.

Reply:
> `customdata` and `ids` are deemed ok

I wouldn't go that far 😏 I just didn't want to hold up the rest of this PR for that fairly minor issue.

Reply:
> But really, at the very least we should add a test checks that `parcoords` traces don't have `hoverlabel` in their fullData.

@etpinard Nice catch! It is addressed in commit da03ceb

Reply:
> I wouldn't go that far  I just didn't want to hold up the rest of this PR for that fairly minor issue.

Ok. I wrote up a summary in https://github.com/plotly/plotly.js/issues/3058, no need to address this now.

That said, I would prefer switching back those `coerceUnlessPruned` calls to regular `coerce` calls for `customdata` and `ids` as these are never pruned at the moment - which could confuse devs in future iterations.

Reply:
That’s fine, easy enough to add back later after I convince you to remove those from some types.

Reply:
Happy to see we can meet halfway :smile: ! Commit 55881c6 calls the regular `coerce` instead of `coerceUnlessPruned`. @etpinard I left the `coerceUnlessPruned` function there in case @alexcjohnson convince you later. Let me know if that's OK